cpmpy/transformations/flatten_model.py

```diff
@@ -675,3 +703,23 @@ def negated_normal(expr):
         #raise NotImplementedError("negate_normal {}".format(expr))
         return expr == 0 # can't do better than this...
 
+def _should_wsum(sub_expr):
```

Comment:
expressions.core already has a _wsum_should and _wsum_make. They are almost identical?

Do you really need this here? If for the '-', then perhaps make a _wsum_should_flatten() that calls _wsum_should as much as possible... same for _wsum_make_flatten (if a new one is really needed...)

Reply:
New one allows for flattening of complex expressions that can be simplified into 1 wsum.